File: algorithm_python/Algorithm_Session/Day3/boj_10828.py
import sys

input = sys.stdin.readline

class Stack:
    def __init__(self):
        self.stk = []
        self.tos = 0

    def push(self, x):
        self.stk.append(x)
        self.tos += 1

    def pop(self):
        if self.tos == 0:
            return -1
        
        self.tos -= 1
        return self.stk[self.tos]

    def size(self):
        return self.tos

    def empty(self):
        return 1 if self.tos == 0 else 0
    
    def top(self):
        if self.tos == 0:
            return -1
        return self.stk[self.tos - 1]

n = int(input())
stk = Stack()
ans = ""

for _ in range(n):
    s = input().split(" ")
    op = s[0]
    
    if op == "push":
        stk.push(int(s[1]))
    elif op == "top":
        ans += str(stk.top())
    elif op == "pop":
        ans += str(stk.pop())
    elif op == "size":
        ans += str(stk.size())
    elif op == "empty":
        ans += str(stk.empty())

    if op != "push":
        ans += "\n"

print(ans)

File: 02_Algorithm/algorithm_java/Week2/Stack/boj_10828.java
package Week2.Stack;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj_10828 {
    static int[] stk;
    static int tos = 0;
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static StringTokenizer st;

    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(br.readLine());
        stk = new int[n];

        while (n > 0) {
            st = new StringTokenizer(br.readLine(), " ");
            int x = -2;

            switch (st.nextToken()) {
                case "push":
                    push(Integer.parseInt(st.nextToken()));
                    break;

                case "pop":
                    x = pop();
                    break;

                case "top":
                    x = top();
                    break;

                case "size":
                    x = size();
                    break;

                case "empty":
                    x = empty();
                    break;
            }

            if (x != -2) sb.append(x).append("\n");
            n--;
        }

        System.out.println(sb);
    }

    static void push(int x) {
        stk[tos++] = x;
    }

    static int pop() {
        return (tos == 0) ? -1 : stk[--tos];
    }

    static int top() {
        return (tos == 0) ? -1 : stk[tos - 1];
    }

    static int empty() {
        return (tos == 0) ? 1 : 0;
    }

    static int size() {
        return tos;
    }
}
